spec/services/make_an_offer_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+RSpec.describe MakeAnOffer do
+  describe 'validation' do
+    it 'accepts nil conditions' do
+      decision = MakeAnOffer.new(
+        application_choice: create(:application_choice, status: :application_complete),
```

Comment:
this way we are creating 4 times the
`create(:application_choice, status: :application_complete`.

I'm in favour of using a let in these cases:

`let(:completed_application_choice) { create(:application_choice, status: :application_complete }`

apart from performance, it is good for readability as well.

Reply:
No strong feelings about the style — please feel free to fix this up if you like. `let`s are not cached across examples, so I don't think there's a performance benefit to be had here.

Reply:
@duncanjbrown thanks for reminding me the about let caching :)
I think @tijmenb suggestion is a good approach.
ps.
I think there are others way to cache the application_choice, (`before :all` for example) but probably at the expense of readability.

Reply:
Thanks @tijmenb: fixed up with `build_stubbed`